### PR TITLE
feat: implement `ssz_generic` spec tests

### LINKED ISSUE #460: Implement `ssz_generic` spec tests
This is the spec test runner: https://github.com/ethereum/consensus-specs/tree/dev/tests/formats/ssz_generic

### PR DESCRIPTION
**Pull request by f3r10**
**Motivation**

Closes #460

### CODE REVIEW

File: lib/ssz_ex.ex [hunk 1]
```diff
@@ -44,8 +44,10 @@ defmodule LambdaEthereumConsensus.SszEx do
   @offset_bits 32
 
   defp encode_int(value, size) when is_integer(value), do: {:ok, <<value::size(size)-little>>}
+  defp encode_int(value, _size), do: {:error, "#{inspect(value)} is not a integer"}
```
[the remark this answers is not in the document]
Comment (f3r10) [reply lib/ssz_ex.ex:47]: I think `catch_error` is the assert I was looking for :sweat_smile:

File: lib/spec/runners/ssz_generic.ex [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    expected =
+      YamlElixir.read_from_file!(case_dir <> "/value.yaml")
+      |> SpecTestUtils.sanitize_yaml()
+
+    assert_ssz("valid", schema, real_deserialized, expected)
+  end
+
+  defp handle_case("invalid", schema, real_deserialized, _testcase) do
+    assert_ssz("invalid", schema, real_deserialized)
+  end
```
Comment (MegaRedHand) [inline lib/spec/runners/ssz_generic.ex:47]: I think the serialized and deserialized values here are flipped:

```suggestion
  defp handle_case("valid", schema, real_serialized, testcase) do
    case_dir = SpecTestCase.dir(testcase)

    real_deserialized =
      YamlElixir.read_from_file!(case_dir <> "/value.yaml")
      |> SpecTestUtils.sanitize_yaml()

    assert_ssz("valid", schema, real_serialized, real_deserialized)
  end

  defp handle_case("invalid", schema, real_serialized, _testcase) do
    assert_ssz("invalid", schema, real_serialized)
  end
```

File: lib/ssz_ex.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (MegaRedHand) [reply lib/ssz_ex.ex:47]: In fact, the problem is this call here:
https://github.com/lambdaclass/lambda_ethereum_consensus/pull/490/files#diff-62d8fda618c9c6bc05df1a37a282bd1aee3c33a9f37bcb309cc94f47ca0f5b4dR59

It should be `SszEx.decode`, not `SszEx.encode`.

File: lib/spec/runners/helpers/ssz_static_containers/var_test_struct.ex [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  """
+  @behaviour LambdaEthereumConsensus.Container
+
+  fields = [
+    :A,
+    :B,
+    :C
+  ]
+
+  @type uint16 :: 0..unquote(2 ** 16 - 1)
```
Comment (MegaRedHand) [inline lib/spec/runners/helpers/ssz_static_containers/var_test_struct.ex:13]: This should be in `SszTypes`

File: lib/spec/runners/helpers/ssz_static_containers/small_test_struct.ex [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  Struct definition for `SmallTestStruct`.
+  """
+  @behaviour LambdaEthereumConsensus.Container
+
+  fields = [
+    :A,
+    :B
+  ]
+
+  @type uint16 :: 0..unquote(2 ** 16 - 1)
```
Comment (MegaRedHand) [inline lib/spec/runners/helpers/ssz_static_containers/small_test_struct.ex:12]: Same

File: lib/spec/runners/ssz_generic.ex [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        #         ["uint" <> size, max_size | _] ->
+        # {:vector, {:int, String.to_integer(size)}, String.to_integer(max_size)}
+        #       end
+        #   end
+        # "bitlist" ->
+        #   case cse do
+        #     "bitlist_" <> rest ->
+        #       [size | _] = String.split(rest, "_")
+        #       {:bitlist, :bool, String.to_integer(size)}
+        #   end
```
Comment (MegaRedHand) [inline lib/spec/runners/ssz_generic.ex:113]: This should have a TODO

**Review by MegaRedHand — APPROVED** [review]
Good work 🔥

**Pull request merged**